### ISSUE TITLE
nix-eshell doesn't play well with flake.nix

### ISSUE THREAD

**Issue by SohamG** [NixOS/nix-mode#164]
Title

I have a flake.nix at the root of my project that defines a `devShell`. From reading the source for nix-eshell, its trying to `nix-instantiate` the file its given, and I don't think that works well with flakes.

Am I missing an obvious solution?

**Comment by natto1784** [comment]
yep it still needs devShell support, for now it just `nix shell`s. Something I would like to see as well.

**Comment by gustavomedeiross** [comment]
if someone has a problem running devShells in eshell, I highly recommend trying the eshell integration from https://codeberg.org/akib/emacs-eat, worked perfectly for me
